test/e2e/providers/azure/azure.go

```diff
@@ -31,38 +31,21 @@ const (
 // Provider is a provider struct for testing Azure
 type Provider struct {
 	oc     *clusterinfo.OpenShift
+	status *config.InfrastructureStatus
 	vmSize string
 }
 
 // New returns a new Azure provider struct with the give client set and ssh key pair
-func New(clientset *clusterinfo.OpenShift, hasCustomVXLANPort bool) (*Provider, error) {
-	if hasCustomVXLANPort == true {
-		return nil, fmt.Errorf("custom VXLAN port is not supported on current Azure image")
-	}
-
+func New(clientset *clusterinfo.OpenShift, status *config.InfrastructureStatus) *Provider {
 	return &Provider{
 		oc:     clientset,
+		status: status,
 		vmSize: defaultVMSize,
-	}, nil
+	}
 }
 
 // newAzureMachineProviderSpec returns an AzureMachineProviderSpec generated from the inputs, or an error
-func newAzureMachineProviderSpec(clusterID string, status *config.PlatformStatus, location, zone, vmSize string) (*mapi.AzureMachineProviderSpec, error) {
-	if clusterID == "" {
-		return nil, fmt.Errorf("clusterID is empty")
-	}
-	if status == nil || status == (&config.PlatformStatus{}) {
-		return nil, fmt.Errorf("platform status is nil")
-	}
-	if status.Azure == nil || status.Azure == (&config.AzurePlatformStatus{}) {
-		return nil, fmt.Errorf("azure platform status is nil")
-	}
-	if status.Azure.NetworkResourceGroupName == "" {
-		return nil, fmt.Errorf("azure network resource group name is empty")
-	}
-	rg := status.Azure.ResourceGroupName
-	netrg := status.Azure.NetworkResourceGroupName
-
+func (p *Provider) newAzureMachineProviderSpec(location, zone string) (*mapi.AzureMachineProviderSpec, error) {
```

Comment:
Now that you are reworking this, I suggest naming this `getAzureMachineProviderSpect`

Reply:
The function is actually constructing a new AzureMachineProviderSpec based on the input, to me it feels the `new` prefix fits better

Reply:
I see your point. Leave it as is then.